lib/board.rb

```diff
@@ -0,0 +1,53 @@
+class Board
+    attr_accessor :cells
+
+    def initialize
+        reset!
+    end
+
+    def reset!
+        @cells = Array.new(9, " ")
+    end
+
+    def display
+        puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+        puts "-----------"
+        puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+        puts "-----------"
+        puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+    end
+
+    def position(input)
+        index = input.to_i - 1
+        @cells[index]
+    end
+
+    def full?
+        @cells.all? do |slot|
+            slot == "X" || slot == "O"
+        end
+    end
+
+    def turn_count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```